addons/website_sale_loyalty/controllers/main.py

```diff
@@ -77,6 +77,8 @@ def claim_reward(self, reward, **post):
         for coupon, rewards in claimable_rewards.items():
             if reward_id in rewards:
                 coupon_id = coupon
+        if(reward_id.program_type == 'coupons' or reward_id.program_type == 'gift_card'):
+            return self.pricelist(coupon_id.code)
```

Comment:
I don't understand this change honestly. Why only those two?
```suggestion
        if (reward_id.program_type == 'coupons' or reward_id.program_type == 'gift_card'):
            return self.pricelist(coupon_id.code)
```

EDIT: ok I understand, it's because the code isn't applied yet.
```suggestion
        if (coupon_id not in order.applied_coupon_ids):
            # The reward was presented via a linked loyalty.card but needs further checks.
            return self.pricelist(coupon_id.code)
```